### PR TITLE
Mlflow log

### PR DESCRIPTION
**Pull request by SKholkin**
MLflow is very convinient tracking system with UI (which DO NOT affect on TensorBoard logging)
With many opportunities for managing experiments
For example MLflow UI allows to easily navigate through experiments and separate runs
image

### CODE REVIEW

**Comment by SKholkin** [comment]
Another important feature of MLflow is runs logging by Params and Metrics and you can sort runs using them.
image
Also you can filter runs by Params and Metrics using SQL like clauses
image

**Comment by SKholkin** [comment]
Work in progress. Now MLflow logging enabled only in classification sample but soon it will be easily integrated in another samples (like object-detection).

**Review by ljaljushkin — CHANGES_REQUESTED** [review]

**Comment by ljaljushkin** [comment]
Could you also attach a screenshot of how to filter not finished experiments?

File: examples/classification/main.py [hunk 1]
```diff
@@ -250,6 +264,7 @@ def train(config, compression_ctrl, model, criterion, is_inception, lr_scheduler
 
             for key, value in stats.items():
                 if isinstance(value, (int, float)):
+                    mlflow.log_metric("compression/statistics/{0}".format(key), value, step=len(train_loader) * epoch)
```
Comment (vanyalzr) [inline examples/classification/main.py:267]: Step numbers in iterations are going to be very large for datasets like ImageNet. Maybe add a flag to write steps as epoch counts.

**Review by ljaljushkin — CHANGES_REQUESTED** [review]

File: examples/common/utils.py [hunk 2]
```diff
@@ -97,6 +99,15 @@ def configure_paths(config):
 def configure_logging(sample_logger, config):
     config.tb = SummaryWriter(config.log_dir)
 
+    if config.mode.lower() == 'train' and config.to_onnx is None:
```
Comment (ljaljushkin) [inline examples/common/utils.py:102]: move this condition to function in common/utils.py `is_mlflow_logging_enabled`

**Comment by ljaljushkin** [comment]
please resolve merge conflicts and fix failed tests in pre-commit validation

**Comment by SKholkin** [comment]
> Could you also attach a screenshot of how to filter not finished experiments?

image

You can filter by both 'FINISHED' and 'UNFINISHED' attributes

File: examples/classification/main.py (hunk 1 above)
Comment (SKholkin) [reply examples/classification/main.py]: Done

**Comment by ljaljushkin** [comment]
@SKholkin, please click on "Ready for review" button to convert PR from draft type to the ordinary one, when you feel that everything is ready for review and merging

**Review by ljaljushkin — APPROVED** [review]
Hope we will be able to reduce the number of dependencies coming with the MLFlow package by using [mlflow-client package in future](https://github.com/mlflow/mlflow/issues/2557)

**Comment by SKholkin** [comment]
No need to merge all commits, please squish them.

File: examples/classification/common.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        cudnn.benchmark = False
+
+
+def load_resuming_checkpoint(resuming_checkpoint_path):
+    resuming_model_sd = None
+    resuming_checkpoint = None
+    if resuming_checkpoint_path is not None:
+        resuming_model_sd, resuming_checkpoint = load_resuming_model_state_dict_and_checkpoint_from_path(
+            resuming_checkpoint_path)
+    return resuming_model_sd, resuming_checkpoint
```
Comment (vshampor) [inline examples/classification/common.py:32]: What is the purpose of this file? What different code locations are these functions "common" between?

**Review by vshampor — CHANGES_REQUESTED** [review]

File: examples/classification/common.py (hunk 3 above)
Comment (SKholkin) [reply examples/classification/common.py:32]: This file combines some "common" parts of code from `main_worker ` (classification/main.py) and `staged_quantization_main_worker` (classification/staged_quantization_worker.py) . `main_worker` was way too big for going through pylint so it was splitted using this file. There may be possible "common" parts  further in classification/main.py and classification/staged_quantization_worker.py so it classification/common.py can also be extended later.

**Review by vshampor — APPROVED** [review]

**Pull request merged**